tests/jballerina-unit-test/src/test/resources/test-src/statements/assign/assign-stmt.bal

```diff
@@ -116,6 +116,16 @@ function assignAnyToUnionWithErrorAndAny() {
     assertEquality(4, y);
 }
 
+function testAssignVarInQueryExpression() {
+    xml x1 = xml `<book>The Lost World</book>`;
+
+    var x2 = from xml element in x1 select element.toBalString();
+    assertTrue(x2 is string[]);
+
+    var x3 = from xml element in x1 select element;
+    assertTrue(x3 is xml[]);
+}
```

Comment:
Shall we add a test case to assert the output of this query expression?

Reply:
I will do it.

Reply:
Updated.8f225565d0cb19cd953a142da1c38498beefe3a7